schema/deploy/mutations/generate_reports.sql

```diff
@@ -0,0 +1,66 @@
+-- Deploy cif:mutations/generate_reports to pg
+
+begin;
+
+create or replace function cif.generate_reports(revision_id int, report_type text, start_date timestamptz, end_date timestamptz)
+  returns setof cif.form_change as $generate_reports$
+  declare
+  form_change_record record;
+  start_date_year int := extract(year from $3);
+  report_interval_start_date timestamptz;
+  report_interval interval := '1 year';
+  initial_index int := 1;
+  temp_row record;
+begin
+  if $3 >= $4 then
+      raise exception 'start_date must be before end_date';
+
+  elsif $2 = 'Annual' then
+    -- if start date is before or equal to jan 31st, we will start the first annual report on the same year otherwise we will start on the next year
+    if $3 <= make_date(start_date_year, 01, 31) then
+      report_interval_start_date := make_date(start_date_year, 01, 31);
+    else
+      report_interval_start_date := make_date(start_date_year + 1, 01, 31);
+    end if;
+
+  elsif $2 = 'Quarterly' then
+    report_interval := '3 month'::interval;
+    -- choosing the closest quarter start date from a list of possible dates as a temporary table
+    select * from generate_series(make_date(start_date_year, 01, 05), make_date(start_date_year + 1, 01, 05), report_interval) as temp_table(quarterly_report_due_date)
+      where $3 <= quarterly_report_due_date limit 1 into report_interval_start_date;
```

Comment:
no need for a complex alias here
```suggestion
    select * from generate_series(make_date(start_date_year, 01, 05), make_date(start_date_year + 1, 01, 05), report_interval) as quarterly_report_due_date
      where $3 <= quarterly_report_due_date limit 1 into report_interval_start_date;
```